sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -321,6 +320,14 @@ class GpuParquetWriter(
     }
   }
 
+  override def transform(batch: ColumnarBatch): ColumnarBatch = {
+    val transformedCols = GpuColumnVector.extractColumns(batch).safeMap { cv =>
+      new GpuColumnVector(cv.dataType, deepTransformColumn(cv.getBase, cv.dataType))
```

Comment:
`deepTransformColumn` needs some comments (not an issue in this pr). I assume with the name that the input is not closed. With this latest patch, I think we need to close `batch` after we close `deepTransformColumn` on all the columns.